packages/format-library/src/inline-typography/index.js

```diff
@@ -0,0 +1,334 @@
+import { inlineTypography } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { getTypographyFontSizeValue } from '../../../block-editor/src/components/global-styles/typography-utils';
+
+// Utility functions
+const serializeStyle = ( styleObj ) => {
+	return (
+		Object.entries( styleObj )
+			.filter( ( [ , value ] ) => value )
+			.map(
+				( [ key, value ] ) =>
+					key.replace( /([A-Z])/g, '-$1' ).toLowerCase() + ':' + value
+			)
+			.join( ';' ) + ';'
+	);
+};
+
+const parseStyle = ( styleString ) => {
+	const styleObj = {};
+	if ( styleString ) {
+		styleString.split( ';' ).forEach( ( style ) => {
+			const [ key, value ] = style.split( ':' );
+			if ( key && value ) {
+				const camelKey = key
+					.trim()
+					.replace( /-([a-z])/g, ( match, letter ) =>
+						letter.toUpperCase()
+					);
+				styleObj[ camelKey ] = value.trim();
+			}
+		} );
+	}
+	return styleObj;
+};
+
+// Constants
+const TEXT_TRANSFORM_OPTIONS = [
+	{ label: __( 'None' ), value: '' },
+	{ label: __( 'Uppercase' ), value: 'uppercase' },
+	{ label: __( 'Lowercase' ), value: 'lowercase' },
+	{ label: __( 'Capitalize' ), value: 'capitalize' },
+];
+
+const name = 'core/inline-typography';
+const title = __( 'Inline Typography' );
+
+const Edit = ( { isActive, value, onChange, contentRef } ) => {
+	const [ showSettings, setShowSettings ] = useState( false );
+	const activeFormat = getActiveFormat( value, name );
+	const originalFontSize = activeFormat?.attributes?.fontSize;
+	const computedStyles = useMemo( () => {
+		return activeFormat && activeFormat.attributes.style
+			? parseStyle( activeFormat.attributes.style )
+			: {};
+	}, [ activeFormat ] );
+
+	const {
+		fontFamily = '',
+		letterSpacing = '',
+		textTransform = '',
+		lineHeight = '',
+	} = computedStyles;
+	const themeFontFamiliesSetting = useSettings( 'typography.fontFamilies' );
+	// Check if it's an array and if its first element contains a "theme" property.
+	const themeFontFamilies = useMemo( () => {
+		if (
+			Array.isArray( themeFontFamiliesSetting ) &&
+			themeFontFamiliesSetting.length > 0 &&
+			themeFontFamiliesSetting[ 0 ].theme
+		) {
+			return themeFontFamiliesSetting[ 0 ].theme;
+		}
+		return [];
+	}, [ themeFontFamiliesSetting ] );
+	// If themeFontFamilies exist, use them; otherwise, use a static fallback.
+	const dynamicFontFamilyOptions = useMemo( () => {
+		const options = [ { label: __( 'Default' ), value: '' } ];
+		if ( themeFontFamilies.length > 0 ) {
+			themeFontFamilies.forEach(
+				( { name: themeName, fontFamily: themeFont } ) => {
+					options.push( {
+						label: themeName,
+						value: themeFont,
+					} );
+				}
+			);
+		} else {
+			options.push(
+				{ label: 'Arial', value: 'Arial, sans-serif' },
+				{ label: 'Helvetica', value: 'Helvetica, sans-serif' },
+				{ label: 'Georgia', value: 'Georgia, serif' },
+				{
+					label: 'Times New Roman',
+					value: '"Times New Roman", Times, serif',
+				}
```

Comment:
Why are these chosen as the defaults?
You can find a list of the default fonts here:
https://github.com/WordPress/gutenberg/blob/trunk/packages/base-styles/_variables.scss#L15

These labels also needs to be translation ready, since these are only latin characters.